### ISSUE TITLE
Pylint raises `signature-differs` on overridden `__new__` with optional arguments in superclass constructor.

### ISSUE THREAD

**Issue by sg495** [pylint-dev/pylint#8780]
### Bug description

The following code raises `signature-differs` for `B.__new__`:

```py
# pylint: disable = too-few-public-methods, missing-docstring

from typing import Self

class A:
    x: int
    def __new__(cls, x: int = 0) -> Self:
        instance = super().__new__(cls)
        instance.x = x
        return instance

class B(A):
    y: str
    def __new__(cls, y: str) -> Self:
        instance = super().__new__(cls, 0)
        instance.y = y
        return instance

```

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
test.py:14:4: W0222: Signature differs from overridden '__new__' method (signature-differs)
```

### Expected behavior

This warning is raised inconsistently: it happens when there are optional arguments in the constructor of the superclass, but it disappears when the default value is removed (even though the signature still differs). In my opinion, either all incompatible overloads of `__new__` should raise this warning, or none should. Previous discussion on the Mypy github (cf. https://github.com/python/mypy/issues/14836) seems to indicate that the latter would be the preferred option.

(If the superclass constructor has variadic keyword arguments, `arguments-differ` is raised instead.)

### Pylint version

```shell
pylint 2.17.1
astroid 2.15.1
Python 3.11.2 (tags/v3.11.2:878ead1, Feb  7 2023, 16:38:35) [MSC v.1934 64 bit (AMD64)]
```

### OS / Environment

Windows 10

**Comment by jacobtylerwalls** [comment]
Hi @sg495 :wave:

pylint's concept behind `signature-differs` is to check "can you substitute the child class for calls to the base class signature", in other words, it's checking for LSP. 1-argument calls will succeed on the base class, fail on the subclass.

This is essentially the [paradigmatic example for this check].
